### PR TITLE
DT-569: First stab at moving integration tests away from gradle

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -11,6 +11,7 @@ plugins {
   id 'org.owasp.dependencycheck' version '5.3.0'
   id 'com.github.ben-manes.versions' version '0.28.0'
   id 'org.jetbrains.kotlin.jvm' version '1.3.61'
+  id 'com.gorylenko.gradle-git-properties' version "2.2.2"
```
Comment (petergphillips) [inline build.gradle]: add in git version too

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/SwaggerConfig.java [hunk 2]
```diff
@@ -48,17 +41,7 @@ public Docket offenderApi() {
...
-        BuildProperties buildProperties;
-        try {
-            buildProperties = (BuildProperties) applicationContext.getBean("buildProperties");
-        } catch (final BeansException be) {
-            final var properties = new Properties();
-            properties.put("version", "?");
-            buildProperties = new BuildProperties(properties);
-        }
-
+    private ApiInfo apiInfo(final BuildProperties buildProperties) {
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/config/SwaggerConfig.java]: I think this was all needed in previous versions of intellij as it didn't do a gradle build before running so didn't have any properties.  Integration should all now work properly.

File: src/test/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/PingIntTest.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import uk.gov.justice.digital.hmpps.oauth2server.resource.IntegrationTest
+
+class PingIntTest : IntegrationTest() {
+  @Test
+  fun `Ping returns pong`() {
+    webTestClient.get().uri("/auth/ping")
+        .exchange()
+        .expectStatus().isOk
+        .expectHeader().contentType("text/plain;charset=UTF-8")
+        .expectBody(String::class.java).isEqualTo<Nothing>("pong")
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/PingIntTest.kt]: the `Nothing` here is really weird and can't be right - should be `String`.  Apparently some kotlin bug :man_shrugging:

File: src/test/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthorisationIssuerIntTest.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package uk.gov.justice.digital.hmpps.oauth2server.resource.api
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.http.MediaType
+import uk.gov.justice.digital.hmpps.oauth2server.resource.IntegrationTest
+
+class AuthorisationIssuerIntTest : IntegrationTest() {
+  @Test
+  fun `Jwk set page returns JwkSet`() {
```
Comment (mikehalmamoj) [inline src/test/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthorisationIssuerIntTest.kt]: This test is in JwkSetIntTest

File: src/test/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/PublicKeyIntTest.kt [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.test.context.TestPropertySource
+import uk.gov.justice.digital.hmpps.oauth2server.resource.IntegrationTest
+
+@TestPropertySource(properties = ["jwt.jwk.key.id=some-key-id"])
+class PublicKeyIntTest : IntegrationTest() {
+  @Test
+  fun `Jwk set page returns JwkSet`() {
```
Comment (mikehalmamoj) [inline src/test/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/PublicKeyIntTest.kt]: Change description?

File: src/test/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthorisationIssuerIntTest.kt (hunk 4 above)
Comment (petergphillips) [reply src/test/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthorisationIssuerIntTest.kt]: agreed, copy and paste error

File: src/test/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/PingIntTest.kt (hunk 3 above)
Comment (mikehalmamoj) [reply src/test/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/PingIntTest.kt]: I can't run this test locally - `No qualifying bean of type 'org.springframework.boot.info.BuildProperties' available`

**Pull request merged**